src/stream/mod.rs

```diff
@@ -1077,6 +1078,7 @@ where
         };
     }
 
+    assert!(i <= len, "must not underflow");
```

Comment:
These functions are only ever called with 0 or 1 so perhaps instead of the assertion (which may impact performance in the hottest part of a parser). There could be an enum with Zero and One taking as argument to the functions?

```
enum UnconsStart {
    Zero = 0,
    One = 1,
}
// Use like
start as usize
```

Reply:
Good call. I also moved the math to a saturating subtraction. If you can think of a better way to guard against passing an empty slice with `UnconsStart::One` I'm all ears!